bool leg
bool LEFT = 0
bool RIGHT = 1
---
bool result
bool SUCCESS = True
bool FAIL = False
